net/net.c

```diff
@@ -5222,25 +5222,37 @@ static void get_subnet_incomming_syn(host_node_type *host_node_ptr)
 {
     struct sockaddr_in lcl_addr_inet;
     size_t lcl_len = sizeof(lcl_addr_inet);
-    struct hostent *he;
-    int hlen;
-    char *subnet;
-
-    if (!getsockname(host_node_ptr->fd, &lcl_addr_inet,
-                     (socklen_t *)&lcl_len)) {
-        he = gethostbyaddr(&lcl_addr_inet.sin_addr,
-                           sizeof lcl_addr_inet.sin_addr, AF_INET);
-        /*if (gbl_verbose_net)*/
-        fprintf(stderr, "Incoming connection from name: %s (%s:%u)\n",
-                (he && he->h_name) ? he->h_name : "unknown",
+    struct hostent *he = NULL;
+
+    int ret = getsockname(host_node_ptr->fd, &lcl_addr_inet,
+                     (socklen_t *)&lcl_len);
+    if (ret != 0) {
+        logmsg(LOGMSG_ERROR, "Failed to getsockname() for fd=%d\n",
+               host_node_ptr->fd);
+        return;
+    }
+    
+    char host[NI_MAXHOST], service[NI_MAXSERV];
+    int s = getnameinfo((struct sockaddr *) &lcl_addr_inet,
+            lcl_len, host, NI_MAXHOST,
+            service, NI_MAXSERV, NI_NUMERICSERV);
+
+    if (s != 0) {
+        logmsg(LOGMSG_ERROR, "Error from getaddrinfo: %s\n", gai_strerror(s));
+        logmsg(LOGMSG_INFO, "Incoming connection from name: %s (%s:%u)\n",
```

Comment:
C&P of below, extra %s

Reply:
I'm pretty sure i had it printf 'unknown' in some iteration of the fix, and then i switched to logmsg. Thanks for catching.